specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2973 @@
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        }
+      },
+      "description": "Request from client to perform migration assessment"
+    },
+    "MigrationAssessmentResource": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        },
+        "targetPlatformProviderId": {
+          "type": "string",
+          "description": "Target platform(Flexible Server) provider ID"
+        },
+        "targetServerDetails": {
+          "$ref": "#/definitions/TargetServerDetails"
+        },
+        "summary": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssessmentSummaryItem"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "featureName"
+          ],
+          "description": "Represents a summary of Migration Assessment"
+        },
+        "replicaAssessmentReports": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MigrationAssessmentResource"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "sourceServerResourceId"
+          ],
+          "description": "Represents Migration assessment resource of replica servers"
+        },
+        "assessmentReportCompletionDateTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Assessment report completion date time"
+        }
+      },
+      "description": "Represents a resource name availability."
+    },
+    "TargetServerDetails": {
+      "type": "object",
+      "properties": {
+        "serverResource": {
+          "$ref": "#/definitions/Server",
+          "readOnly": true,
+          "description": "Represents resource properties of target server"
+        },
+        "serverParameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Configuration"
+          },
+          "readOnly": true,
+          "description": "Represents server parameters of target server"
+        },
+        "firewallRules": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FirewallRule"
+          },
+          "readOnly": true,
+          "description": "Represents firewall rules of target server"
+        },
+        "platformProviderId": {
```

Comment:
>platformProviderId [](http://example.com/codeflow?start=9&length=18)

Is this also a string constant == "Microsoft.DBforMySQL/flexibleServers"? If so, I'd recommend removing it unless there's a scenario for it that I'm not getting.

Reply:
this is a string constant. It is an identifier for which target platform the assessment is being performed for.